lambda/src/main/scala/pricemigrationengine/migrations/GuardianWeekly2025Migration.scala

```diff
@@ -165,22 +166,37 @@ object GuardianWeekly2025Migration {
 
     val zuora_subscription = subscription
 
-    val subscriptionRatePlanId =
-      "TODO:subscriptionRatePlanId" // this is to be read from the rate plan in the subscription
-    val removeProduct = ZuoraOrdersApiPrimitives.removeProduct(effectDate.toString, subscriptionRatePlanId: String)
-    val triggerDateString = effectDate.toString
-    val productRatePlanId = "TODO:productCatalogueRatePlanId" // read from the product catalogue
-    val chargeOverrides = List(
-      ZuoraOrdersApiPrimitives.chargeOverride("TODO:productRatePlanChargeId", estimatedNewPrice)
-    )
-    val addProduct = ZuoraOrdersApiPrimitives.addProduct(triggerDateString, productRatePlanId, chargeOverrides)
-    val order_subscription = ZuoraOrdersApiPrimitives.subscription(subscriptionNumber, removeProduct, addProduct)
-    val order = ZuoraOrdersApiPrimitives.replace_a_product_in_a_subscription(
-      orderDate.toString,
-      accountNumber,
-      order_subscription
-    )
-
-    Right(order)
+    val order = {
+      for {
+        ratePlan <- SI2025RateplanFromSubAndInvoices.determineRatePlan(subscription, invoiceList)
+        subscriptionRatePlanId = ratePlan.id
+        removeProduct = ZuoraOrdersApiPrimitives.removeProduct(effectDate.toString, subscriptionRatePlanId)
+        triggerDateString = effectDate.toString
+        productRatePlanId = ratePlan.productRatePlanId // We are upgrading on the same rate plan.
+        chargeOverrides = List(
+          ZuoraOrdersApiPrimitives.chargeOverride(
+            ratePlan.ratePlanCharges.headOption.get.productRatePlanChargeId,
+            PriceCap.cappedPrice(oldPrice, estimatedNewPrice, priceCap)
+          )
+        )
+        addProduct = ZuoraOrdersApiPrimitives.addProduct(triggerDateString, productRatePlanId, chargeOverrides)
+        order_subscription = ZuoraOrdersApiPrimitives.subscription(subscriptionNumber, removeProduct, addProduct)
+        o = ZuoraOrdersApiPrimitives.replace_a_product_in_a_subscription(
```

Comment:
```suggestion
        orderPayload = ZuoraOrdersApiPrimitives.replace_a_product_in_a_subscription(
```
naming it something like that might be clearer?

Reply:
Thanks for that. I will update :)

Reply:
Done here: https://github.com/guardian/price-migration-engine/pull/1148/commits/78ac905be648d39aacd2d1c8c2ffc18b8ddf43a6 , thanks!